homeassistant/components/modbus/switch.py

```diff
@@ -260,7 +260,7 @@ def _read_register(self) -> Optional[int]:
 
         self._available = True
 
-        return int(result.registers[0])
+        return bool(result.bits[0] & 1)
```

Comment:
@janiversen I am getting the following error:
```bash
AttributeError: 'ReadHoldingRegistersResponse' object has no attribute 'bits'
2020-10-30 13:38:30 ERROR (MainThread) [homeassistant] Error doing job: Future exception was never retrieved
```

I am using my test branch for `Modbus Light`, where I copied your changed files. Is there anything else, which I am missing here? Perhaps a updated `pymodbus` library?

Reply:
I rebased my code to latest `dev`. Same error.

Reply:
My bad I did not test with python 3.8, I have updated the PR (and added a couple more tests).

Reply:
Can you tell what has changed in python 3.8? Also, shouldn't we still try to convert the value to bool?

Something like:
```python
return bool(result.registers[0] & 1)
```

Let me test the coil switch, too...

Reply:
I wanted to change it to a bool, BUT I ran into problems since you can define state_on and state_off, therefore it is no longer simply the LSB.

I am not sure but the .bits works for coil but not register and both work in python 3.7

Reply:
Did you check the coil switch on 3.8? I'm just asking so that we find a regression if there's any.

Reply:
No I did not, but github runs the test suite on 3.8 (which is how I saw the problem).